langchain/document_loaders/googledrive.py

```diff
@@ -279,23 +286,32 @@ def _load_file_from_id(self, id: str) -> List[Document]:
         done = False
         while done is False:
             status, done = downloader.next_chunk()
-        content = fh.getvalue()
 
-        from PyPDF2 import PdfReader
+        if self.file_loader_cls is not None:
+            fh.seek(0)
+            loader = self.file_loader_cls(file=fh, **self.file_loader_kwargs)
+            docs = loader.load()
+            for doc in docs:
+                doc.metadata["id"] = id
+            return docs
 
-        pdf_reader = PdfReader(BytesIO(content))
-
-        return [
-            Document(
-                page_content=page.extract_text(),
-                metadata={
-                    "source": f"https://drive.google.com/file/d/{id}/view",
-                    "title": f"{file.get('name')}",
-                    "page": i,
-                },
-            )
-            for i, page in enumerate(pdf_reader.pages)
-        ]
+        else:
+            from PyPDF2 import PdfReader
+
+            content = fh.getvalue()
+            pdf_reader = PdfReader(BytesIO(content))
+
+            return [
+                Document(
+                    page_content=page.extract_text(),
+                    metadata={
+                        "source": f"https://drive.google.com/file/d/{id}/view",
```

Comment:
@MthwRobinson do you not want this metadata present as well?

wouldnt it make sense for the different loaders to have at least some common metadata

Reply:
Yeah that definitely makes sense, just missed that in there. I'll get that updated.

Reply:
@hwchase17 - Anything else you wanted to add to this one?